x/fibre/spec.md

```diff
@@ -0,0 +1,221 @@
+  string ip_address = 2;
+}
+```
+
+**Validation Rules:**
+- `validator_address` must be a valid validator operator address
+- `ip_address` must be non-empty and ≤ 45 characters
+- Signer must match the validator address
+- Validator must be in the active set
+
+### MsgRemoveFibreProviderInfo
+
+Allows removal of fibre provider information for validators not in the active set.
+
+```protobuf
+message MsgRemoveFibreProviderInfo {
+  // validator_address is the operator address of the validator to remove
+  string validator_address = 1;
+  // remover_address is the address of the account requesting removal
+  string remover_address = 2;
+}
+```
+
+**Validation Rules:**
+- `validator_address` must be a valid validator operator address
+- `remover_address` must be a valid account address
+- Validator must NOT be in the active set
+- Provider info must exist for the validator
+
+## Events
+
+### EventSetFibreProviderInfo
+
+Emitted when a validator sets or updates their fibre provider information.
+
+```protobuf
+message EventSetFibreProviderInfo {
+  // validator_address is the operator address of the validator
+  string validator_address = 1;
+  // ip_address is the IP address for the fibre service
+  string ip_address = 2;
+}
+```
+
+### EventRemoveFibreProviderInfo
+
+Emitted when fibre provider information is removed.
+
+```protobuf
+message EventRemoveFibreProviderInfo {
+  // validator_address is the operator address of the validator
+  string validator_address = 1;
+  // remover_address is the address of the account that requested removal
+  string remover_address = 2;
+}
+```
+
+## Queries
+
+### QueryFibreProviderInfo
+
+Query fibre provider information for a specific validator.
+
+**Request:**
+```protobuf
+message QueryFibreProviderInfoRequest {
+  // validator_address is the operator address of the validator
+  string validator_address = 1;
+}
+```
+
+**Response:**
+```protobuf
+message QueryFibreProviderInfoResponse {
+  // info contains the fibre provider information
+  FibreProviderInfo info = 1;
+  // found indicates if the validator has registered info
+  bool found = 2;
+}
+```
+
+### QueryAllActiveFibreProviders
+
+Query fibre provider information for all validators in the active set.
+
+**Request:**
+```protobuf
+message QueryAllActiveFibreProvidersRequest {
+  // pagination defines optional pagination parameters
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
+}
+```
+
+**Response:**
+```protobuf
+message QueryAllActiveFibreProvidersResponse {
+  // providers contains all active fibre providers
+  repeated ActiveFibreProvider providers = 1;
+  // pagination defines the pagination response
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+message ActiveFibreProvider {
+  // validator_address is the operator address of the validator
+  string validator_address = 1;
+  // info contains the fibre provider information
+  FibreProviderInfo info = 2;
+}
```

Comment:
Need pub key of validator to verify signatures and voting power

Reply:
this message is strictly for a validator to update the state for their FibreProviderInfo, why do we need the public key or voting power for that? the state machine already has those things.

the state machine already knows the active set, so can verify simply by comparing the valop address and the auth portion of the state machine verifying that the signature is that of the `GetSigners`

there might be some missing context here for how the cosmos-sdk works, likely better to sync to bridge the gaps between our understandings

Reply:
if there is still confusion I'm definitely happy to sync!